docs/resources/tutorials/differential-expression-analysis.md

```diff
@@ -0,0 +1,458 @@
+
+```R
+DGE=DGEList(expression_data,samples=sample_info$sample,group=c(sample_info$sample_info$pop_temp))
+```
+
+### 5. Filtering out lowly expressed genes
+A number of transcripts/genes will be not expressed at all in any sample, or may only be expressed in a small number of samples. In the latter case, testing for differential expression is noisy and under-powered. 
+
+```R
+keep.exprs <- filterByExpr(DGE, group=DGE$samples$group)
+DGE <- DGE[keep.exprs,, keep.lib.sizes=FALSE]
+```
+
+### 6. TMM normalization
+The next step is conducting normalization of the counts across samples so that library size differences and the effects of genes that are highly expressed and sample specific are accounted for. Regarding the latter, we want to avoid having a few genes take up expression sequencing "real estate" given the overall number of reads generated by a sample, such that it reduces the reads in other transcripts in a way that would lead to false positive DE. To do this, we use the trimmed mean of M-values (TMM)  of [Robinson and Oshlack (2010) :octicons-link-external-24:](https://doi.org/10.1186/gb-2010-11-3-r25){:target="_blank"} available in edgeR. Note, one can use other normalization schemes, and I have seen some evidence that conditional quantile normalizartion (CQN) mignt be worth considering as an alternative.
+
+```R
+DGE=calcNormFactors(DGE,method =c("TMM"))
+```
+
+### 7. Examine data for outliers
+We do this to make sure there are no outliers with respect to a particular experimental variable, suggesting a potential issue with an RNA-seq library, and to also check for sample swaps. This can be done with a multidimensional scaling (MDS) plot using the top 500 highest expressed genes. 
+
+We can color-code samples by level of the temperature factor:
+
+```R
+tempvals<-sample_info$temp
+plotMDS(DGE,top=500,col=ifelse(tempvals=="low","blue","red"),gene.selection="common")
+```
+<center>
+    <img src="../../../../img/tutorials/de_temp_mds.png" alt="MDS plot on expression data: temperature" width="75%" />
+</center>
+
+Or by the levels of the population factor:
+
+```R
+popvals<-sample_info$population
+plotMDS(DGE,top=500,col=ifelse(popvals=="maine","darkgreen","dodgerblue"),gene.selection="common")
+```
+<center>
+  <img src="../../../../img/tutorials/de_pop_mds.png" alt="MDS plot on expression data: population" width="75%" />
+</center>
+
+As expected, there is clear separation between temeprature regimes and geographic locations, but no outliers indicating bad samples or potential label swaps.
+
+### 8. Create design matrix
+At the heart of linear modeling are design matrices that specify boolean variables for the intercept, and additional factors, and limma requires such a matrix for performing differential expression analysis. Our first analysis is going to focus on a one-factor model, in which we ignore geography and look at the effects of low and high temperature treatments. We make a design matrix for this analysis as follows:
```

Comment:
i think it is more reasonable to make a subset of the data to just one group first, then do this analysis. otherwise people will get confused about what is a reasonable thing to do. basically make a fake actual one-variable experiment. open to discussion on this point though